### PR TITLE
Add failing test for GetRequest w/ AwsSdk2Transport (#1068)

### PR DESCRIPTION
**Pull request by workmanw**
This adds a failing test for #1068

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -45,6 +45,7 @@ This section is for maintaining a changelog for all breaking changes for the cli
 
 ### Fixed
 - Deserialize aggregation containing a parent aggregation ([#706](https://github.com/opensearch-project/opensearch-java/pull/706))
+- Error deserialization w/ AwsSdk2Transport ([#1068](https://github.com/opensearch-project/opensearch-java/pull/1068))
```
Comment (dblock) [inline CHANGELOG.md:48]: Let's try to be more specific? `Deserialize error response when using AwsSdk2Transport`?

**Review by dblock — COMMENTED** [review]
@Xtansia @reta mind taking a quick look at this one?

File: CHANGELOG.md (hunk 1 above)
Comment (workmanw) [reply CHANGELOG.md]: 👍

File: java-client/src/main/java/org/opensearch/client/transport/aws/AwsSdk2Transport.java [hunk 2]
```diff
@@ -617,6 +633,22 @@ private static <T> Optional<T> or(Optional<T> opt, Supplier<? extends Optional<?
         }
     }
 
+    private static ByteArrayInputStream toByteArrayInputStream(InputStream is) throws IOException {
```
Comment (workmanw) [inline java-client/src/main/java/org/opensearch/client/transport/aws/AwsSdk2Transport.java:636]: I'm not crazy about this solution, but it's effectively what `ApacheHttpClient5Transport` is doing by using `BufferedHttpEntity`.

File: java-client/src/main/java/org/opensearch/client/transport/aws/AwsSdk2Transport.java [hunk 3]
```diff
@@ -617,6 +633,22 @@ private static <T> Optional<T> or(Optional<T> opt, Supplier<? extends Optional<?
...
+        }
+
+        ByteArrayOutputStream baos = new ByteArrayOutputStream();
+        byte[] buffer = new byte[1024];
+        int len;
+        while ((len = is.read(buffer)) > -1) {
+            baos.write(buffer, 0, len);
+        }
+        baos.flush();
+        return new ByteArrayInputStream(baos.toByteArray());
```
Comment (Xtansia) [inline java-client/src/main/java/org/opensearch/client/transport/aws/AwsSdk2Transport.java:649]: We could re-use aws-sdk-java's `IoUtils` to shorten this: https://github.com/aws/aws-sdk-java-v2/blob/master/utils/src/main/java/software/amazon/awssdk/utils/IoUtils.java#L39-L52
```suggestion
        return new ByteArrayInputStream(IoUtils.toByteArray(is));
```

**Review by Xtansia — DISMISSED** [review]
The changes all look reasonable to me and in-line with the HttpClient5 transport's behaviour

File: java-client/src/main/java/org/opensearch/client/transport/aws/AwsSdk2Transport.java (hunk 3 above)
Comment (workmanw) [reply java-client/src/main/java/org/opensearch/client/transport/aws/AwsSdk2Transport.java]: Oh awesome. This definitely removes verbosity 💯 .

File: java-client/src/main/java/org/opensearch/client/transport/aws/AwsSdk2Transport.java [hunk 4]
```diff
@@ -10,10 +10,7 @@
 
 import jakarta.json.JsonObject;
 import jakarta.json.stream.JsonParser;
-import java.io.ByteArrayInputStream;
-import java.io.IOException;
-import java.io.InputStream;
-import java.io.UnsupportedEncodingException;
+import java.io.*;
```
Comment (reta) [inline java-client/src/main/java/org/opensearch/client/transport/aws/AwsSdk2Transport.java:13]: Could you please revert to non-`*` import? (`spotless` should have spotted that)

**Review by reta — APPROVED** [review]

File: java-client/src/main/java/org/opensearch/client/transport/aws/AwsSdk2Transport.java (hunk 4 above)
Comment (workmanw) [reply java-client/src/main/java/org/opensearch/client/transport/aws/AwsSdk2Transport.java]: Done. I think `spotless` actually made this change. My IDE is set to never do wildcards. Either way, it's fixed.

**Review by Xtansia — APPROVED** [review]

**Pull request merged**